### DISCUSSION TITLE
OKD 12 vSphere Install doesn't accept externial IP address for Virtual IP Address for API & Virtual IP Address for Ingress

### DISCUSSION THREAD

**Discussion by johnlongo** [okd-project/okd#1403] [General]
OKD 12 vSphere Install doesn't accept externial IP addresses for Virtual IP Address for API & Virtual IP Address for Ingress.
With OKD 11 I could assign externial IP addresses like 192.168.4.* for Virtual IP Address for API & Virtual IP Address for Ingress.

Is there a way to assign externial IP addresses when creating an OKD 12 vSphere cluster?

**Comment by darren-oxford** [comment]
I have always assumed that the VIPs have to be on the host machineNetwork range, as the default route etc is taken from the machineNetwork config. Are you trying to set the VIP as an address outside the machineNetwork range?

**Reply by johnlongo** [reply]
Yes, in OKD 11, I could specify an IP like 192.168.4.20 & 192.168.4.21 and it worked for me. Matter of fact, I just created the new cluster using those IP addresses to try to upgrade to 12. Wanted to create a pristine cluster (no changes by me) to do the 12 upgrade. so I started with 4.11.0-0.okd-2022-07-29-154152 then upgraded to 4.11.0-0.okd-2022-10-28-153352 and finally upgraded to 4.11.0-0.okd-2022-11-05-030711. But I'm still getting warnings that are stopping me from upgrading to 12 see attached below. Important Note, I did not modify the cluster in anyway except for the OKD updates.


![image](https://user-images.githubusercontent.com/23619505/201709290-d03b950b-6e2f-45f0-bb4a-3ddca60bc30c.png)

![image](https://user-images.githubusercontent.com/23619505/201710343-bbdd0940-8b68-4e05-93f1-1fe295fa51a6.png)

**Reply by darren-oxford** [reply]
I'm not sure I follow, there is no 4.12 in the 4-stable channel at https://origin-release.apps.ci.l2s4.p1.openshiftapps.com

I may be wrong, but I don't think there is an upgrade path to 4-scos-stable

**Reply by johnlongo** [reply]
In the past when I was on a lower version I automatically was allowed to upgrade to the next version (9 -> 10 -> 11), is 12 different? Will I eventually be given a 12 path to upgrade?

**Comment by darren-oxford** [comment]
Yes, you are on the stable 4 channel, the current latest version is 4.11.0-0.okd-2022-11-05-030711.

4.12.0-0.okd-scos-2022-10-25-053756 is a different branch based on CentOS instead of Fedora. There is not yet a stable release 4.12 based on Fedora. As soon as that is released you will see it as an upgrade option.

**Reply by johnlongo** [reply]
Ok ty
